pkg/image/registry/imagestreamtag/rest.go

```diff
@@ -107,8 +118,216 @@ func (r *REST) Get(ctx kapi.Context, id string) (runtime.Object, error) {
 	return newISTag(tag, imageStream, image, false)
 }
 
+func (r *REST) Create(ctx kapi.Context, obj runtime.Object) (runtime.Object, error) {
+	istag, ok := obj.(*imageapi.ImageStreamTag)
+	if !ok {
+		return nil, kapierrors.NewBadRequest(fmt.Sprintf("obj is not an ImageStreamTag: %#v", obj))
+	}
+	if err := rest.BeforeCreate(Strategy, ctx, obj); err != nil {
+		return nil, err
+	}
+	namespace, ok := kapi.NamespaceFrom(ctx)
+	if !ok {
+		return nil, kapierrors.NewBadRequest("a namespace must be specified to import images")
+	}
+
+	// lifting source and hint information from annotations.  These will be dropped on a future get, so that's a thing
+	tagSource := istag.Annotations["source"]
```

Comment:
@deads2k have a look at [TagReference](https://github.com/openshift/origin/blob/16375903659e3b820233ffe16e1528a27a31b0a6/pkg/image/api/v1/types.go#L78-L92) which is part of [ImageStreamTag](https://github.com/openshift/origin/blob/16375903659e3b820233ffe16e1528a27a31b0a6/pkg/image/api/v1/types.go#L179).